plasma_framework/contracts/src/vaults/Vault.sol

```diff
@@ -5,12 +5,16 @@ import "../framework/PlasmaFramework.sol";
 import "../framework/utils/Operated.sol";
 
 contract Vault is Operated {
+    event SetDepositVerifierCalled(address nextDepositVerifier);
     PlasmaFramework framework;
     bytes32[16] zeroHashes;
 
-    address private _currentDepositVerifier;
-    address private _newDepositVerifier;
-    uint256 private _newDepositVerifierEffectivePeriod;
+    /**
+     * @notice Stores deposit verifier contracts addresses where first was effective upto
+     *  `newDepositVerifierMaturityTimestamp` point of time and second become effective after
+    */
+    address[2] public depositVerifiers;
+    uint256 public newDepositVerifierMaturityTimestamp = 2 ** 256 - 1; // point far in the future
```

Comment:
does this even work and not overflow?

Reply:
I tested it and behaves as intended, but it may be causing confusions in the future.
`2 ** 255` is safe as well, but does not require thinking and testing the overflow.

Reply:
Yeah, I was confused by it as well :+1: